### PR TITLE
BraveSessionCache: Use std::mt19937 random source.

### PR DESCRIPTION
**Pull request by rillian**
Replace the custom linear feedback shift register with the standard
library prng used elsewhere in the file. Less custom code is better.

According to Mark Pilgrim there are no crypto or performance
implications to the change of source.

Resolves 

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/third_party/blink/renderer/core/execution_context/execution_context.cc [hunk 1]
```diff
@@ -40,17 +34,15 @@ float ConstantMultiplier(double fudge_factor, float value, size_t index) {
 }
 
 float PseudoRandomSequence(uint64_t seed, float value, size_t index) {
-  static uint64_t v;
+  static std::mt19937_64 prng;
```
[the remark this answers is not in the document]
Comment (iefremov) [reply chromium_src/third_party/blink/renderer/core/execution_context/execution_context.cc:37]: Hey @rillian , I'm not the code owner - just randomly noticed this PR in the list.
Random generators  from `<random>` are banned by the Chromium C++ policy, so we actually should be replacing them with something from `base/rand_util.h` - see https://chromium.googlesource.com/chromium/src/+/refs/changes/74/3123974/15/styleguide/c++/c++11.md#Random-Number-Engines

there are many other C++-specific peculiarities, checking out https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++ might be helpful

[the remark this answers is not in the document]
Comment (rillian) [reply chromium_src/third_party/blink/renderer/core/execution_context/execution_context.cc:37]: Thanks for the review! I'll rework it to use `base::RandUint64`. That performs a syscall, so may be too slow for `AudioFarblingCallback`. We should probably use either `base::InsecureRandomGenerator` (allegedly 400x faster!) or modify the call site the process the whole buffer at once.

[the remark this answers is not in the document]
Comment (rillian) [reply chromium_src/third_party/blink/renderer/core/execution_context/execution_context.cc:37]: On further investigation, `base::RandUint64` won't work here.

We're farbling `BraveSessionCache` objects, so we explicitly need a repeatable sequence with deterministic keying. From what I read, the Chromium style prohibition is about preventing accidental use of weak/slow `std` algorithms where the lack of entropy would be important.

**Comment by rillian** [comment]
Closing in favour of #10091 which has better CI support.

**Pull request closed without merging**